[DO-392] Define links as a key concept (#135)
th2-net/th2-docs: content/docs/versions/1-7/fundamentals/key-concepts.md

A _pin_ is a point of connection to a box. _Pins_ are required to create channels of communication between boxes. A _pin_ can be used for synchronous and asynchronous messaging depending on its connection type.

_Pins_ required by a module are declared in the module's custom resource by the user. Every declared _pin_ has a `name` and a `connection-type`. 

## Links
A _link_ is a connection between two th2 components.

> It can describe the directional communication between two pins or the relationship between a box and dictionary.

Reviewer: ```suggestion
It can describe the directional communication between two pins or the relationship between a box and a dictionary.
```

> A th2 box can be linked to multiple boxes, dictionaries, or a combination of both.

Reviewer: ```suggestion
th2 box can be linked to the multiple boxes, dictionaries, or the combination of both.
```

> _Link’s_ allow th2 users to easily organize the intended relationships within their test environment.

Reviewer: ```suggestion
_Links_ allow th2 users to easily organize the intended relationships within their test environment.
```

> _Link’s_ allow th2 users to easily organize and clearly communicate to the infra operator the intended relationships within their test environment.

Reviewer 2: Mentioning infra operator looks like some background information is missing. So it is better to delete it.

```suggestion
_Link’s_ allow th2 users to easily organize the intended relationships within their test environment.
```

Each individual _link_ is a simple declaration and can be identified by its name.

> All _links_ are declared in a single custom resource of kind `Th2Link`.

Reviewer: ```suggestion
All _links_ are declared in a single custom resource (CR) of kind `Th2Link`.
```

> This custom resource is located in its dedicated folder , aptly named links, in the infra-schema.

Reviewer: ```suggestion
This custom resource is located in its dedicated folder ,aptly named links, in the infra-schema.
```

> This custom resource is located in its dedicated folder ,aptly named links, in the infra-schema.

Reviewer 2: `,aptly` - it looks strange

Author: so the reason I used this word is because the Links CR is in a folder called Links 
we can have it in a folder called L, or inks, but we don't 
so it is kind of a mixture of obviousness and appropriateness

aptly : in a manner that is appropriate or suitable in the circumstances.

Reviewer 2: OK then change position of `,` only

```suggestion
This custom resource is located in its dedicated folder, aptly named links, in the infra-schema.
```
